### DISCUSSION TITLE
Getting started with Nadine as a Hackspace/Makerspace member management portal.

### DISCUSSION THREAD

**Discussion by proffalken** [nadineproject/nadine#459] [Q&A] [answered]
Hey all,

I posted this on the #nadineproject slack channel as well, but that seems to be *very* quiet so I figured I may as well duplicate things here!

I'm part of the team behind [Make Monmouth](https://www.makemonmouth.co.uk/) and our original plan to use Tendenci for our website and membership management portal really isn't going well.

I searched around for alternatives and found Nadine, and given how quick it was to spin up a local dev version I'm interested in learning much more about it.

Specifically, I'm wondering:

   * If there is an explanation of the various fields when creating a new membership type (I've no idea what the "allowance" and "overage rate" equate to
   * How I integrate with Stripe to take payments (there's a variable in the settings file, and a PR saying it's enabled, but nothing in the docs

In all other aspects this looks great, so hopefully it will help us manage our membership more effectively than Tendenci which just doesn't seem to work for the most basic of things!

**Comment by jsayles** [comment, marked as answer]
Hello and I'm excited you found Nadine.  I'm talking with another maker space that is thinking of using Nadine and I'll direct them to chime in here with their stories.  As for your questions:
1) Documentation would be wonderful as it is very lacking at the moment.  Honestly it would be quite a project to flesh something out and really make it usable.  I could answer a few questions but that's far from designing a good user manual.  Do you have write access to the wiki here?  Feel free to start a page and I'll fill in the blanks.  

1) Stripe is *kinda* integrated and your milage will very.  I don't use stripe so the hooks are probably not really what you want.  That said, this should be something that gets cleaned up as it is the most common payment gateway.

**Reply by proffalken** [reply]
Thanks, happy to document as long as I can get a guiding hand!

Happy to dive into the code as well - would be good to know how it should work at the moment, as setting my payment provider to Stripe in the admin backend causes the code to throw an error because it can't find a particular view.

I've got a reasonable amount of python experience, not much wih Django though, so happy to chip in on the code as well - love the concept, and would far rather work on something that already exists than reinvent the wheel!

**Reply by proffalken** [reply]
@jsayles - first main question, when you talk about "overage rate", what do you mean exactly? I think this might be a US-ism as it doesn't appear to have a UK equivalent :(

At a guess, it's the charge for any additional days that are incurred once the allowance has been used up?

**Reply by jsayles** [reply]
You are correct and it is about the resource not the location. Your membership gives you access to resources, some are all in and some have a set amount of use (allowance).  If you use more than that you are charged the overage rate for each use.  This is known as a Trackable Resource.  An example is coworking days or a 5 day per month membership. Other resources are not tracked, such as mail or a desk and allowance and overage doesn't apply. 

We'll have to think about how/if we need to restructure the resources for a maker space.

**Reply by proffalken** [reply]
Ok, thanks.

For now I think that's fine, as we can just set the allowance to 31 days/month for all membership types - it's possible we might want to make use of the overage feature in future to track who is using the space and how often, so don't worry about it too much!

I'll try and find time to add a line to the docs about this today.

**Reply by proffalken** [reply]
Whilst looking into this as well as stripe integration, I found that https://github.com/nadineproject/nadine/commit/b39c51ef2552509cf1921ba972750924c3c8e3ad appears to have removed the support for Stripe.

I've added those lines back in, and I can connect to the Stripe checkout, but I don't appear to be able to actually setup a payment for it to be collected?

Am I missing something else that should have been added back in?

**Comment by trickfilm** [comment]
Hi proffalken,
I've been emailing Jacob about a similar issue and he suggested I contact you here.
I'm part of the ops team at the [Fimbel Maker & Innovation Lab](https://www.mtholyoke.edu/fimbellab), the makerspace at Mt Holyoke College in Massachusetts, US. We are using way too many different tools to manage users, equipment, schedules, training, etc, and I'm looking for an open-source, extensible platform that will do everything we need. Nadine looked like it might fit the bill, with the right customizations.
We also haven't ruled out the possibility of developing our own system and making it freely available as an open source project.
If you're interested in either of these ideas, let's talk!

**Reply by proffalken** [reply]
If you're happy to I'd rather continue the discussion here so that anyone else who might be in the same position can join in as well?

I'd love to see Nadine get the features we're looking for, and at some point I'm hoping to link it to [my inventory system](https://GitHub.com/makemonmouth/mventory) (that also needs further development!) Aas it would be awesome to have the inventory updated based on someone's login!

**Reply by trickfilm** [reply]
sure that's fine too!

**Reply by proffalken** [reply]
Great!

I'd be interested in knowing what's top of your list as far as functionality is concerned?

I'm mainly interested in getting Stripe payments up and running, then focusing on other things, but I'm getting an error when adding in my Stripe config so if you've already done this I'd love to hear how!

**Reply by trickfilm** [reply]
Overage isn't a big deal for us, as an academic makerspace. But we do need a good way to track student training, authorizations for equipment, maintenance & incident logs, and scheduling of people, rooms, and tools. Right now we have to use 4 separate calendar systems and it's making us crazy.
[Here's a description](https://docs.google.com/document/d/193PP7Hn9BdMpOSMCY__h6gr-3lJMkx6yc2prWPZgi9c/edit?usp=sharing) of my proposed system.

**Reply by jsayles** [reply]
I don't think the memberships are the way to track trainings.  If I understand it correctly, a training is something like "this person was trained on the table saw and now has access to the resource".  Is that correct?  Memberships only make sense in the context of time as in "This user has access to the coworking desks starting here and ending there".  A training has no end date so I think we need to think of other ways to get this data in the system. 

That said, since your space isn't charging folks you might be able to get away with just creating memberships and never ending them.   I don't really like that though as everything in Nadine is built around the idea of "These people are members" and that requires an in and an out.  It isn't just a constraint of Nadine, it is how communities work.
